From 6c5eba69149d681f81a83a7770a6ad8ab662d583 Mon Sep 17 00:00:00 2001
From: David Henningsson <david.henningsson@canonical.com>
Date: Thu, 12 Jan 2012 10:58:34 +0100
Subject: [PATCH 2/8] alsa-mixer: Allow speaker port to control "Front
 Speaker"

"Front Speaker", "Surround Speaker" seems to be a common enough name
to make it into alsa-utils, so we should probably care about it as
well. In this case, there was a macbook pro whose speakers didn't work
without these controls.

BugLink: http://bugs.launchpad.net/bugs/551441
Reported-by: Jeroen T. Vermeulen <jtv@canonical.com>
Signed-off-by: David Henningsson <david.henningsson@canonical.com>
---
 .../alsa/mixer/paths/analog-output-speaker.conf    |   30 +++++++++++++++++++-
 1 files changed, 29 insertions(+), 1 deletions(-)

diff --git a/src/modules/alsa/mixer/paths/analog-output-speaker.conf b/src/modules/alsa/mixer/paths/analog-output-speaker.conf
index c6916d6..0ba3e91 100644
--- a/src/modules/alsa/mixer/paths/analog-output-speaker.conf
+++ b/src/modules/alsa/mixer/paths/analog-output-speaker.conf
@@ -50,7 +50,7 @@ switch = mute
 volume = zero
 
 [Element Speaker]
-required = any
+required-any = any
 switch = mute
 volume = merge
 override-map.1 = all
@@ -66,6 +66,13 @@ volume = merge
 override-map.1 = all-front
 override-map.2 = front-left,front-right
 
+[Element Front Speaker]
+switch = mute
+volume = merge
+override-map.1 = all-front
+override-map.2 = front-left,front-right
+required-any = any
+
 [Element Rear]
 switch = mute
 volume = merge
@@ -78,6 +85,13 @@ volume = merge
 override-map.1 = all-rear
 override-map.2 = rear-left,rear-right
 
+[Element Surround Speaker]
+switch = mute
+volume = merge
+override-map.1 = all-rear
+override-map.2 = rear-left,rear-right
+required-any = any
+
 [Element Side]
 switch = mute
 volume = merge
@@ -90,10 +104,24 @@ volume = merge
 override-map.1 = all-center
 override-map.2 = all-center,all-center
 
+[Element Center Speaker]
+switch = mute
+volume = merge
+override-map.1 = all-center
+override-map.2 = all-center,all-center
+required-any = any
+
 [Element LFE]
 switch = mute
 volume = merge
 override-map.1 = lfe
 override-map.2 = lfe,lfe
 
+[Element LFE Speaker]
+switch = mute
+volume = merge
+override-map.1 = lfe
+override-map.2 = lfe,lfe
+required-any = any
+
 .include analog-output.conf.common
-- 
1.7.8.3

